src/server.ts

```diff
@@ -235,52 +238,114 @@ app.post("/pdf/authenticate", async (c) => {
+
+    return c.json({
+      authenticated: authCheck.authenticated,
+    });
+  } catch (error) {
+    console.error("Error checking session authentication status:", error);
+    return c.json({ error: "Internal server error" }, 500);
+  }
+});
+
 // PDF Upload URL Route (for presigned uploads)
 app.post("/pdf/upload-url", async (c) => {
   try {
     const { sessionId, fileName, fileSize } = await c.req.json();
 
+    console.log("Upload URL request received for sessionId:", sessionId);
+    console.log("FileName:", fileName);
+    console.log("FileSize:", fileSize);
+
     if (!sessionId || !fileName) {
+      console.log("Missing sessionId or fileName");
       return c.json({ error: "sessionId and fileName are required" }, 400);
     }
 
     // Check if session is authenticated
     const sessionIdObj = c.env.SessionFileTracker.idFromName(sessionId);
     const sessionTracker = c.env.SessionFileTracker.get(sessionIdObj);
 
+    console.log(
+      "Created session tracker for upload URL, sessionId:",
+      sessionId
+    );
+
     const authCheckResponse = await sessionTracker.fetch(
       "https://dummy-host/is-session-authenticated",
       {
         method: "GET",
       }
     );
 
+    console.log(
+      "Auth check response status for upload URL:",
+      authCheckResponse.status
+    );
+
     const authCheck = (await authCheckResponse.json()) as {
       authenticated: boolean;
     };
+
+    console.log("Auth check result for upload URL:", authCheck);
+
     if (!authCheck.authenticated) {
+      console.log("Session not authenticated for upload URL, returning 401");
       return c.json({ error: "Session not authenticated" }, 401);
     }
 
+    console.log("Session authenticated, generating upload URL");
+
     // Generate unique file key
     const fileKey = `uploads/${sessionId}/${crypto.randomUUID()}-${fileName}`;
 
     // Generate direct upload URL to R2 bucket
     // This creates a URL that uploads directly to R2, bypassing the worker
     const uploadUrl = `/pdf/upload/${fileKey}`;
 
+    console.log("Generated fileKey:", fileKey);
+    console.log("Generated uploadUrl:", uploadUrl);
+
     // Add file metadata to SessionFileTracker
-    await sessionTracker.fetch("https://dummy-host/add-file", {
-      method: "POST",
-      headers: { "Content-Type": "application/json" },
-      body: JSON.stringify({
-        sessionId,
-        fileKey,
-        fileName,
-        fileSize: fileSize || 0,
-        metadata: { status: "uploading" },
-      }),
-    });
+    const addFileResponse = await sessionTracker.fetch(
+      "https://dummy-host/add-file",
+      {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({
+          sessionId,
+          fileKey,
+          fileName,
+          fileSize: fileSize || 0,
+          metadata: { status: "uploading" },
+        }),
+      }
+    );
+
+    console.log("Add file response status:", addFileResponse.status);
```

Comment:
this is another file that's getting a bit out of hand size-wise :). happy to let you merge as long as you are willing to trade off and wait for me to clean up/break down after merges :)